### PR TITLE
Checked whether let and const are used properly and then changed accordingly

### PR DESCRIPTION
**Pull request by gauravdas014**
issue  #1195

### CODE REVIEW

File: prework/week2LoopsExercise.js [hunk 1]
```diff
@@ -39,10 +39,10 @@
 // run without ever stopping). Then fix the loop so that it stops when
 // counterFour is equal to -100.
 
-let counterFour = 1;
+const counterFour = 1;
 while (counterFour < 2) {
-    console.log('HELP ME!')
-    counterFour--;
+  console.log('HELP ME!')
+  counterFour--;
```
Comment (alodahl) [inline prework/week2LoopsExercise.js]: this needs to be `let`, as `counterFour` is changed on line 45.

Comment (gauravdas014) [reply prework/week2LoopsExercise.js]: updated

File: functional-programming/FP.md [hunk 2]
```diff
@@ -50,13 +50,13 @@ The purpose of `map` is to process every element of an array in the exact same w
...
 ```JS
-    let intArray = [1, 2, 3]
+    const intArray = [1, 2, 3]
 
     function add3(value) {
       return value + 3;
     }
 
-    let result = intArray.map(add3) // result is [4, 5, 6]
+    let const = intArray.map(add3) // result is [4, 5, 6]
```
[the remark this answers is not in the document]
Comment (gauravdas014) [reply functional-programming/FP.md]: Ma'am when I click on the edit option it shows "const result" but when I click on the view it shows "let const", I don't know why is this happening.

**Comment by vegetabill** [comment]
Looks good overall. 

@gauravdas014 what searching regex did you use to find these? I'm seeing a bunch that weren't found using this search: `ack 'let [a-z]+ ='`

for example, it finds:
```
functional-programming/FP.md
82:    let total = seasonTwoEpisodes.reduce(addDuration, 0)
```

along with a few others. Can you try to find these and fix them in additional commit(s).

Thanks!

**Comment by gauravdas014** [comment]
@vegetabill Updated

**Comment by vegetabill** [comment]
@gauravdas014 Thanks for fixing those but I'm still seeing some unhandled. Did you use my example ack command above?

Anyway here is a list of what I'm still seeing present on this PR branch:

```
javascript/javascript-maps.md
66:     let things = new Map();
109:    let animals = new Set();

javascript/javascript-7-oop.md
136:> let books = [];

runtime-complexity/runtime1-analyzing.js
28:  let primes = []

testing-and-tdd/mocha-testing.md
141:let foo = "foo";
153:let foo = "foo";
165:let foo = "foo";
```

File: functional-programming/FP.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ The purpose of the filter is to return a specified subset of the original array
 Example: Let's say that you have an array of restaurants, and you want to know what your options are for only restaurants that are open at 7a. You have a class called `Restaurant`, which has a property `openingTime`.
```
Comment (Kundan28) [inline functional-programming/FP.md]: There's a typo to at `7a`

Comment (gauravdas014) [reply functional-programming/FP.md]: type fixed

**Comment by gauravdas014** [comment]
@vegetabill Updated. Sir, actually I don't know much about the regex.  I used the manual search of 'let' and 'const' in my code editor. I know the method I used is never a good approach. But now I get to know that we can use regex to find something in our directory. So from now on, I will be using regex, Thank you!

**Comment by vegetabill** [comment]
Looks great thanks!

**Comment by Kundan28** [comment]
Nice work @gauravdas014 
Updated in scoresheet too.

**Pull request merged**